modules/cockroachdb/cockroachdb.go

```diff
@@ -0,0 +1,111 @@
+package cockroachdb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/url"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultSQLPort   = "26257"
+	defaultAdminPort = "8080"
+
+	defaultImage     = "cockroachdb/cockroach"
+	defaultImageTag  = "latest-v23.1"
+	defaultDatabase  = "defaultdb"
+	defaultStoreSize = "100%"
+)
+
+// CockroachDBContainer represents the CockroachDB container type used in the module
+type CockroachDBContainer struct {
+	testcontainers.Container
+	opts Options
+}
+
+// MustConnectionString panics if the address cannot be determined.
+func (c *CockroachDBContainer) MustConnectionString(ctx context.Context) string {
+	addr, err := c.ConnectionString(ctx)
+	if err != nil {
+		panic(err)
+	}
+	return addr
+}
+
+// ConnectionString returns the dial address to open a new connection to CockroachDB.
+func (c *CockroachDBContainer) ConnectionString(ctx context.Context) (string, error) {
+	mappedport, err := c.MappedPort(ctx, defaultSQLPort+"/tcp")
+	if err != nil {
+		return "", err
+	}
+
+	hostIP, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	params := url.Values{
+		"sslmode": []string{"disable"},
+	}
+
+	u := url.URL{
+		Scheme:   "postgres",
+		User:     url.User("root"),
+		Host:     net.JoinHostPort(hostIP, mappedport.Port()),
+		Path:     c.opts.Database,
+		RawQuery: params.Encode(),
+	}
+
+	return u.String(), nil
+}
+
+// RunContainer creates an instance of the CockroachDB container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*CockroachDBContainer, error) {
+	req := testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			ExposedPorts: []string{
+				defaultSQLPort + "/tcp",
+				defaultAdminPort + "/tcp",
+			},
+			WaitingFor: wait.ForHTTP("/health").WithPort(defaultAdminPort),
+		},
+		Started: true,
+	}
+
+	// apply options
+	o := defaultOptions()
+	for _, opt := range opts {
+		if apply, ok := opt.(Option); ok {
+			apply(&o)
+		}
+		opt.Customize(&req)
+	}
+
+	req.Image = image(req, o)
+	req.Cmd = cmd(o)
+
+	// start
+	container, err := testcontainers.GenericContainer(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	return &CockroachDBContainer{Container: container, opts: o}, nil
+}
+
+func image(req testcontainers.GenericContainerRequest, opts Options) string {
+	if req.Image != "" {
+		return req.Image
+	}
+	return fmt.Sprintf("%s:%s", defaultImage, opts.ImageTag)
+}
+
+func cmd(opts Options) []string {
+	return []string{
+		"start-single-node",
+		"--insecure",
```

Comment:
Given support for USERNAME, PASSWORD, DATABASE env vars are already available in recent versions, those should be supported for the module. See [implementation](https://github.com/testcontainers/testcontainers-java/pull/6448) in Java.

Reply:
Tags on Cockroach images have various versions that aren't going to be semver compatible. I'm wondering if you'd be happy if we always set the env vars, they just won't take effect with versions `<22.1`. That seems easier than trying to support tags like:

- `latest-v23.1`
- `v23.1.13-fips`

WDYT?

Reply:
Cockroach seems happier when I drop `--insecure` & add `--accept-sql-without-tls`. Will see about landing a change here

Reply:
Pushed the a change to include changing user & password that will set the necessary env vars & connection string